tests/set_test.go

```diff
@@ -21,3 +34,57 @@ func TestSet(t *testing.T) {
 		}
 	}
 }
+
+/*
+We open some connections to the db and fire concurrent SET commands for a particular key
+with different values. We expect that there are no dirty reads/writes.
+All the values read should be among the values that were attempted to set in the first place.
+*/
+func TestConcurrentSetCommands(t *testing.T) {
+	numOfConnections := 4
+	connectionValues := make(map[*net.Conn]*string)
+	expectedValues := make(map[string]struct{})
+	valuesReadChan := make(chan interface{}, numOfConnections)
+
+	// Create connections and the values to set through them.
+	for connNum := 0; connNum < numOfConnections; connNum++ {
+		value := GenerateRandomString(8, "abcdefghizklmopqrs12345")
+		connectionValues[getLocalConnectionPtr()] = value
+		expectedValues[*value] = struct{}{}
+	}
+
+	// Execute the SET commands from the connections, and pass the output of GET to a channel
+	var wgroup sync.WaitGroup
+	key := "sample_key"
+	for conn, value := range connectionValues {
+		wgroup.Add(1)
+		go executeCommands(conn, &key, value, valuesReadChan, &wgroup)
+	}
+	wgroup.Wait()
+	fmt.Println("Received values from all connections")
+	close(valuesReadChan)
+
+	// Verify the values received in the channel
+	assert.Equal(t, numOfConnections, len(valuesReadChan))
+	for valueRead := range valuesReadChan {
+		if valueRead != nil {
+			valueReadStr := valueRead.(string)
+			_, ok := expectedValues[valueReadStr]
+			if !ok {
+				fmt.Println("Value read is not in expected values' map")
+				t.Fail()
+				break
+			}
+		}
+	}
+}
+
+func executeCommands(conn *net.Conn, key, value *string, valReadChan chan interface{}, wGroup *sync.WaitGroup) {
+	defer wGroup.Done()
+	defer (*conn).Close()
+	fmt.Println("Goroutine started")
```

Comment:
nit: Are these `fmt.Println` statements required? Please remove if was added for debug purposes

Reply:
Removed.